### PR TITLE
Hotfix/account provider should be lowercase

### PR DESCRIPTION
**Pull request by emrysal**
## What does this PR do?

* `Account.provider` is 'google' whereas iDP was 'GOOGLE', - `account.provider` has the right lowercase value.
* Updates the identityProviderId to the account.providerAccountId instead of the userId
* Create a userland solution for fixing this retroactively.

### CODE REVIEW

**Comment by emrysal** [comment]
Hi @deepakprabhakara Setting the identityProviderId to the userId seems very deliberate, but the ID we get from Google for example is very, very different. How was this intended to work and are you happy with this change?

File: apps/web/pages/auth/logout.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default function Logout(props: Props) {
           </div>
         </div>
       </div>
-      <Button href="/auth/login" passHref className="flex w-full justify-center">
```
Comment (emrysal) [inline apps/web/pages/auth/logout.tsx:46]: This is an unrelated fix for an error in the console.

File: packages/features/auth/lib/next-auth-options.ts [hunk 2]
```diff
@@ -537,6 +537,33 @@ export const AUTH_OPTIONS: AuthOptions = {
           },
         });
 
+        /* --- START FIX LEGACY ISSUE WHERE 'identityProviderId' was accidentally set to userId --- */
```
Comment (keithwillcode) [inline packages/features/auth/lib/next-auth-options.ts:540]: @emrysal Should we write a migration to fix this for all users in 1 go?

**Review by keithwillcode — APPROVED** [review]

File: packages/features/auth/lib/next-auth-options.ts (hunk 2 above)
Comment (emrysal) [reply packages/features/auth/lib/next-auth-options.ts:540]: Opted against it as I'm not sure we have the identityProviderId for all Google accounts, looks like the account system is newer - we should revisit in a ~2 months and drop this code.
